planner/plannertest/cmp.go

```diff
@@ -0,0 +1,114 @@
+package plannertest
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/flux/planner"
+	"github.com/influxdata/flux/semantic/semantictest"
+)
+
+// CompareLogicalPlans recursively compares two logical query plans.
+// It returns an error if it encounters a non-matching node.
+// TODO(jlapacik): Need a way to consistently order nodes
```

Comment:
We should define a `Walk` method similar to Specs.  Edges are ordered, so there's no reason not to have a deterministic traversal order.

Reply:
Edges of logical plan are ordered?